packages/utils/src/stableMath.ts

```diff
@@ -0,0 +1,209 @@
+import { BigNumber } from "bignumber.js";
+
+const BIG_ZERO = new BigNumber(0)
+const BIG_ONE = new BigNumber(1)
+const BIG_TEN = new BigNumber(10)
+
+
+
+// Computes the invariant given the current balances, using the Newton-Raphson approximation.
+// The amplification parameter equals: A n^(n-1)
+export const _calculateInvariant = (
+  amplificationParameter: BigNumber,
+  balances: BigNumber[]
+) :BigNumber => {
```

Comment:
see first comment re: types -- return value should be parsed to an integer string, or we should use the ethers big number library

Reply:
If this function will just be used internally (i.e. no need for the router to call `_calculateInvariant` directly, you can leave the return type but not export the function